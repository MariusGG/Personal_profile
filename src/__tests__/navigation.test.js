import React from 'react';
import { shallow } from 'enzyme';
import Navigation from '../nav/navigation';

describe ('Navigation', () =>{

  let wrapper

  beforeAll(() =>{
    wrapper = shallow(<Navigation />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a nav component', () => {
    const nav = wrapper.find('nav')
    expect(nav.length).toEqual(1)
  })

  it('renders four Routes', () => {
    const routes = wrapper.find('Route')
    expect(routes.length).toEqual(4)
  })

})
