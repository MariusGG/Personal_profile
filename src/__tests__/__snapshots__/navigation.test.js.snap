// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <nav
          className="nav-links"
        >
          <span
            className="initial"
          >
            MG
          </span>
          <p
            className="name"
          >
            Marius (G) Guei
          </p>
          <ul>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
            >
              Home
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/about"
            >
              About
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/contact"
            >
              Contact
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/projects"
            >
              Projects
            </NavLink>
          </ul>
        </nav>,
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/about"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/contact"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/projects"
        />,
      ],
      "className": "nav-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="initial"
            >
              MG
            </span>,
            <p
              className="name"
            >
              Marius (G) Guei
            </p>,
            <ul>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/"
              >
                Home
              </NavLink>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/about"
              >
                About
              </NavLink>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/contact"
              >
                Contact
              </NavLink>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/projects"
              >
                Projects
              </NavLink>
            </ul>,
          ],
          "className": "nav-links",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MG",
              "className": "initial",
            },
            "ref": null,
            "rendered": "MG",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Marius (G) Guei",
              "className": "name",
            },
            "ref": null,
            "rendered": "Marius (G) Guei",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/"
                >
                  Home
                </NavLink>,
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/about"
                >
                  About
                </NavLink>,
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/contact"
                >
                  Contact
                </NavLink>,
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/projects"
                >
                  Projects
                </NavLink>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Home",
                  "to": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "About",
                  "to": "/about",
                },
                "ref": null,
                "rendered": "About",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Contact",
                  "to": "/contact",
                },
                "ref": null,
                "rendered": "Contact",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Projects",
                  "to": "/projects",
                },
                "ref": null,
                "rendered": "Projects",
                "type": [Function],
              },
            ],
            "type": "ul",
          },
        ],
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/about",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/contact",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/projects",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <nav
            className="nav-links"
          >
            <span
              className="initial"
            >
              MG
            </span>
            <p
              className="name"
            >
              Marius (G) Guei
            </p>
            <ul>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/"
              >
                Home
              </NavLink>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/about"
              >
                About
              </NavLink>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/contact"
              >
                Contact
              </NavLink>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/projects"
              >
                Projects
              </NavLink>
            </ul>
          </nav>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/about"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/contact"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/projects"
          />,
        ],
        "className": "nav-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="initial"
              >
                MG
              </span>,
              <p
                className="name"
              >
                Marius (G) Guei
              </p>,
              <ul>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/"
                >
                  Home
                </NavLink>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/about"
                >
                  About
                </NavLink>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/contact"
                >
                  Contact
                </NavLink>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/projects"
                >
                  Projects
                </NavLink>
              </ul>,
            ],
            "className": "nav-links",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MG",
                "className": "initial",
              },
              "ref": null,
              "rendered": "MG",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Marius (G) Guei",
                "className": "name",
              },
              "ref": null,
              "rendered": "Marius (G) Guei",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/"
                  >
                    Home
                  </NavLink>,
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/about"
                  >
                    About
                  </NavLink>,
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/contact"
                  >
                    Contact
                  </NavLink>,
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/projects"
                  >
                    Projects
                  </NavLink>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "Home",
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "About",
                    "to": "/about",
                  },
                  "ref": null,
                  "rendered": "About",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "Contact",
                    "to": "/contact",
                  },
                  "ref": null,
                  "rendered": "Contact",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "Projects",
                    "to": "/projects",
                  },
                  "ref": null,
                  "rendered": "Projects",
                  "type": [Function],
                },
              ],
              "type": "ul",
            },
          ],
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/about",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/contact",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/projects",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
